File: src/secondary/yuhoToCatala/ast.ml
type literal =
  | LInt of int
  | LFloat of float
  | LString of string
  | LBoolean of bool

type typ =
  | TInt
  | TFloat
  | TString
  | TBoolean
  | TMoney
  | TDate
  | TDuration

type expr =
  | Var of string
  | Lit of literal
  | UnaryOp of string * expr
  | BinaryOp of string * expr * expr

type stmt =
  | VariableDeclaration of typ * string * expr
  | FunctionDeclaration of typ * string * (typ * string) list * expr
  | Scope of string * stmt list
  | Struct of string * (typ * string) list
  | Assertion of expr

type program = stmt list

File: src/archive/v1/transpiler/yuhoToWhiley/ast.rs
#[derive(Debug)]
pub enum Literal {
    Int(i64),
    Float(f64),
    String(String),
    Boolean(bool),
}

#[derive(Debug)]
pub enum Type {
    TInt,
    TFloat,
    TString,
    TBoolean,
    TMoney,
    TDate,
    TDuration,
}

#[derive(Debug)]
pub enum Expr {
    Var(String),
    Lit(Literal),
    UnaryOp(String, Box<Expr>),
    BinaryOp(String, Box<Expr>, Box<Expr>),
}

#[derive(Debug)]
pub enum Stmt {
    VariableDeclaration(Type, String, Expr),
    FunctionDeclaration(Type, String, Vec<(Type, String)>, Expr),
    Scope(String, Vec<Stmt>),
    Struct(String, Vec<(Type, String)>),
    Assertion(Expr),
}

pub type Program = Vec<Stmt>;
